modules/common/systemd/base.nix

```diff
@@ -0,0 +1,295 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  # Ghaf systemd config
+  cfg = config.ghaf.systemd;
+
+  # Override minimal systemd package configuration
+  package =
+    pkgs.systemdMinimal.override {
+      pname = cfg.withName;
+      withAcl = true;
+      withAnalyze = cfg.withDebug;
+      inherit (cfg) withApparmor;
+      inherit (cfg) withAudit;
+      withCompression = true;
+      withCoredump = cfg.withDebug || cfg.withMachines;
+      inherit (cfg) withCryptsetup;
+      inherit (cfg) withEfi;
+      withBootloader = cfg.withEfi; # systemd-boot fails if not explicity set
+      inherit (cfg) withFido2;
+      inherit (cfg) withHostnamed;
+      withImportd = cfg.withMachines;
+      withKexectools = false;
```

Comment:
This is kind of potential minor opportunity to smooth development in `-debug` builds because `kexec` enables `nixos-anywhere` and could potentially allow re-running the ghaf installer directly without rebooting to installer. Take it with "I'm not sure if this kind of functionality could be added to the ghaf installer. If yes, this could be `withKexectools = cfg.withDebug`

Reply:
I had a discussion with @TanelDettenborn over lunch on possibilities to quickly build and test kernel changes without modifying them by building to nix-store and updating boot entries everytime. `kexec` was the only thing that came to my mind so I'd keep that `cfg.withDebug` to support development.

Reply:
Ok will add that to debug